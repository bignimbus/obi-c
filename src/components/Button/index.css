.button,
.button[disabled] {
  appearance: none;
  background-color: transparent;
  border: 2px solid var(--theme-contrast-color);
  color: var(--theme-contrast-color);
  display: block;
  height: 100%;
  margin: 0 auto;
  padding: 0;
  position: relative;
  width: 100%;
}

.button--disabled {
  border-color: transparent;
}

.button__content-container {
  bottom: 50%;
  position: absolute;
  transform: translateY(50%);
  width: 100%;
}
