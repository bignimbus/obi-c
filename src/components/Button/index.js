import './index.css';
import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ disabled, children, onClick }) => (
  <button
    onClick={onClick}
    className={`
      button
      ${disabled ? 'button--disabled' : ''}
    `}
    disabled={disabled}
  >
    <div className='button__content-container'>
      { children }
    </div>
  </button>
);

Button.propTypes = {
  disabled: PropTypes.bool,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default Button;
