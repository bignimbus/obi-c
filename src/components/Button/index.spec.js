import React from 'react';
import renderer from 'react-test-renderer';
import Button from '.';

describe('Button', () => {
  it('should render a button with content and require an onClick event handler', () => {
    const fn = jest.fn();
    const element = renderer.create(
      <Button onClick={fn}>Click me</Button>,
    );
    element.root.findByType('button').props.onClick();
    expect(fn).toHaveBeenCalled();
    expect(element.toJSON()).toMatchSnapshot();
  });

  it('should be able to be disabled', () => {
    const fn = jest.fn();
    const element = renderer.create(
      <Button disabled onClick={fn}>Don't click me</Button>,
    );
    expect(element.toJSON()).toMatchSnapshot();
  });
});
