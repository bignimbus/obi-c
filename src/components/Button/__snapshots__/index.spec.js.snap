// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should be able to be disabled 1`] = `
<button
  className="
      button
      button--disabled
    "
  disabled={true}
  onClick={[MockFunction]}
>
  <div
    className="button__content-container"
  >
    Don't click me
  </div>
</button>
`;

exports[`Button should render a button with content and require an onClick event handler 1`] = `
<button
  className="
      button
      
    "
  onClick={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="button__content-container"
  >
    Click me
  </div>
</button>
`;
